### PR TITLE
Added concurrent queue

### LINKED ISSUE #594: [Feature Request] Add concurrent queue
## 🚀 Feature Request
Concurrent queue

### Is your feature request related to a problem? Please describe.
A concurrent queue is a data structure that allows multiple threads to safely enqueue and dequeue items without causing race conditions or inconsistencies

### Describe the solution you'd like
Add concurrent queue

### Describe alternatives you've considered
NA

### Additional context
NA
---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
### Description
Added concurrent queue

Fixes #594

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Queue/concurrent-queue.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Concurrent Queue Data Structure
```
Comment (ajay-dhangar) [inline docs/Queue/concurrent-queue.md:10]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/Queue/concurrent-queue.md (hunk 1 above)
Comment (mehul-m-prajapati) [reply docs/Queue/concurrent-queue.md:10]: @ajay-dhangar : okay added it.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**